integration-tests/init.spec.js

```diff
@@ -36,22 +36,45 @@ describe('init.js', () => {
     return sandbox.remove()
   })
 
-  context('when dd-trace is not in the app dir', () => {
-    const NODE_OPTIONS = `--require ${path.join(__dirname, '..', 'init.js')}`
-    it('should initialize the tracer, if no DD_INJECTION_ENABLED', () => {
-      return runTest(cwd, { NODE_OPTIONS }, 'true\n')
+  context('preferring app-dir dd-trace', () => {
+    context('when dd-trace is not in the app dir', () => {
+      const NODE_OPTIONS = `--require ${path.join(__dirname, '..', 'init.js')}`
+      it('should initialize the tracer, if no DD_INJECTION_ENABLED', () => {
+        return runTest(cwd, { NODE_OPTIONS }, 'true\n')
+      })
+      it('should not initialize the tracer, if DD_INJECTION_ENABLED', () => {
+        return runTest(cwd, { NODE_OPTIONS, DD_INJECTION_ENABLED }, 'false\n')
+      })
     })
-    it('should not initialize the tracer, if DD_INJECTION_ENABLED', () => {
-      return runTest(cwd, { NODE_OPTIONS, DD_INJECTION_ENABLED }, 'false\n')
+    context('when dd-trace in the app dir', () => {
+      const NODE_OPTIONS = '--require dd-trace/init.js'
+      it('should initialize the tracer, if no DD_INJECTION_ENABLED', () => {
+        return runTest(cwd, { NODE_OPTIONS }, 'true\n')
+      })
+      it('should initialize the tracer, if DD_INJECTION_ENABLED', () => {
+        return runTest(cwd, { NODE_OPTIONS, DD_INJECTION_ENABLED }, 'true\n')
+      })
     })
   })
-  context('when dd-trace in the app dir', () => {
-    const NODE_OPTIONS = '--require dd-trace/init.js'
-    it('should initialize the tracer, if no DD_INJECTION_ENABLED', () => {
-      return runTest(cwd, { NODE_OPTIONS }, 'true\n')
+  context('runtime version check', () => {
+    const NODE_OPTIONS = `--require ${path.join(__dirname, 'init', 'setversion.js')} --require dd-trace/init.js`
+    context('when node version is less than engines field', () => {
+      const FAKE_VERSION = '3.0.0'
+      it('should initialize the tracer, if no DD_INJECTION_ENABLED', () => {
+        return runTest(cwd, { NODE_OPTIONS, FAKE_VERSION }, 'true\n')
+      })
+      it('should not initialize the tracer, if DD_INJECTION_ENABLED', () => {
+        return runTest(cwd, { NODE_OPTIONS, DD_INJECTION_ENABLED, FAKE_VERSION }, 'false\n')
+      })
     })
-    it('should initialize the tracer, if DD_INJECTION_ENABLED', () => {
-      return runTest(cwd, { NODE_OPTIONS, DD_INJECTION_ENABLED }, 'true\n')
+    context('when node version is more than engines field', () => {
+      const FAKE_VERSION = '50000.0.0'
```

Comment:
make sure to update this in april of the year 27002, when node 50000 releases

Reply:
Maybe I should use `Number.MAX_SAFE_INTEGER` instead?

Reply:
![IT'S OVER 9000!!!!](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExMjY5cDhka2s3dDd5OGYwcDNlazZhbjhocGpsbzA2cDBkNmI5d2lvaCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/5xjbWDIgEZSgM/giphy.gif)